doc/api/timers.md

```diff
@@ -2,90 +2,142 @@
 
     Stability: 3 - Locked
 
-All of the timer functions are globals.  You do not need to `require()`
-this module in order to use them.
+The `timer` module exposes a global API for scheduling functions to
+be called at some future period of time. Because the timer functions are
+globals, there is no need to call `require('timers')` to use the API.
 
-## clearImmediate(immediateObject)
+The timer functions within Node.js implement a similar API as the timers API
+provided by Web Browsers but use a different internal implementation that is
+built around [the Node.js Event Loop][].
 
-Stops an `immediateObject`, as created by [`setImmediate`][], from triggering.
+## Class: Immediate
 
-## clearInterval(intervalObject)
+This object is created internally and is returned from [`setImmediate()`][]. It
+can be passed to [`clearImmediate()`][] in order to cancel the scheduled
+actions.
 
-Stops an `intervalObject`, as created by [`setInterval`][], from triggering.
+## Class: Timeout
 
-## clearTimeout(timeoutObject)
+This object is created internally and is returned from [`setTimeout()`][] and
+[`setInterval()`][]. It can be passed to [`clearTimeout`][] or
+[`clearInterval()`][] (respectively) in order to cancel the scheduled actions.
 
-Prevents a `timeoutObject`, as created by [`setTimeout`][], from triggering.
+By default, when a timer is scheduled using either [`setTimeout`] or
+[`setInterval()`][], the Node.js event loop will continue running as long as the
+timer is active. Each of the `Timeout` objects returned by these functions
+export both `timeout.ref()` and `timeout.unref()` functions that can be used to
+control this default behavior.
 
-## ref()
+### timeout.ref()
 
-If a timer was previously `unref()`d, then `ref()` can be called to explicitly
-request the timer hold the program open. If the timer is already `ref`d calling
-`ref` again will have no effect.
+When called, requests that the Node.js event loop *not* exit so long as the
+timer is active. Calling `timeout.ref()` multiple times will have no effect.
 
-Returns the timer.
+Returns a reference to the `Timeout`.
 
-## setImmediate(callback[, arg][, ...])
+### timeout.unref()
 
-Schedules "immediate" execution of `callback` after I/O events'
-callbacks and before timers set by [`setTimeout`][] and [`setInterval`][] are
-triggered. Returns an `immediateObject` for possible use with
-[`clearImmediate`][]. Additional optional arguments may be passed to the
-callback.
+When called, the `Timeout` object (while active) will not require the Node.js
+event loop to remain active. Calling `timout.unref()` multiple times will have
```

Comment:
I'm not sure that's any better. Rather than hold this up while we get the wording perfect on this, I'd recommend landing and iterating on it separately in subsequent PRs